joern-cli/frontends/solidity2cpg/src/main/scala/io/joern/solidity2cpg/passes/AstCreator.scala

```diff
@@ -341,52 +327,39 @@ class AstCreator(filename: String, sourceUnit: SourceUnit, global: Global) exten
 
         mAst
       case x =>
-        logger.warn(s"Unhandled statement of tyvar modifierMethod = Ast()pe ${x.getClass}")
+        logger.warn(s"Unhandled statement of type ${x.getClass}")
         Ast() // etc
     }
 
   }
 
   // TODO: I assume the only types coming into parameter are var decls but it's worth making sure in tests
   private def astForParameter(varDecl: VariableDeclaration, order: Int): Ast = {
-    val NewMethodParameter = NewMethodParameterIn();
-    var typefullName       = ""
-    var code               = ""
-    var visibility         = ""
-    var storage            = ""
-    typefullName = getTypeName(varDecl.typeName, varDecl.storageLocation)
-
-    varDecl.visibility match {
-      case x: String => visibility = " " + x
-      case _         => visibility = ""
-    }
+    var storage = ""
 
-    varDecl.storageLocation match {
-      case x: String => storage = " " + x
-      case _         => storage = ""
+    // TODO: I assume the original match exists because of a possible null value. Fix that during parsing.
```

Comment:
A note on this, if you have a deeper look into the JSON parser where these case classes are being constructed you could potentially re-define how some of these missing values are handled there already.